core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -105,6 +105,12 @@ public void delete(Delete delete) {
 
   private Optional<TransactionResult> getFromStorage(Get get) throws CrudException {
     try {
+      // only get after image columns
+      get.clearProjections();
+      LinkedHashSet<String> afterImageColumnNames =
+          tableMetadataManager.getTransactionalTableMetadata(get).getAfterImageColumnNames();
+      get.withProjections(afterImageColumnNames);
+
```

Comment:
Here, I added projections for after image columns to filter out the before image columns.